module/size-speed-range/foundry/ranges.js

```diff
@@ -0,0 +1,180 @@
+
+const simplifiedRangeTable = RangeBandTable.table().map(row => ({
+    moddesc: `${row.band} range (${row.maxDistanceYards} yds)`,
+    max: row.maxDistanceYards,
+    penalty: row.penalty,
+    desc: row.description
+}));
+
+const basicSetRangeTable = SSRT.speedRangeTable(16).map(row => ({
+    moddesc: `for speed/range ${row.yards} yds`,
+    max: row.yards,
+    penalty: row.penalty,
+    desc: `${row.yards} yds`
+}));
+// change the last entry description from 700 yards to 500+ yards
+basicSetRangeTable.forEach(row => {
+    if (row.penalty === -15) {
+        row.moddesc = 'for speed/range 500+ yds';
+        row.max = '500+';
+        row.desc = '500+ yds';
+    }
+});
+
+/*
+  Defines the range strategy used throughout the application. A range strategy
+  is defined as an ordered (closest range to farthest range) array of range 
+  bands. A range band is defined as a structure like this:
+
+  {
+    moddesc: <String: text to use in the modifier bucket>,
+    max: <num: number of yards that define the maximum distance of this band>,				
+    penalty: <num: modifier to use for ranged attacks in this band>,
+    desc: <String: text that describes the range>
+  }
+
+  Responsibilities:
+
+  - Defines a world setting to set the range strategy. Currently supported: 
+    * Standand (Size and Speed/Range Table from Basic).
+    * Simplifed (Range bands from Monster Hunters 2: The Enemy).
+   
+  - On update of the setting, update the modifier bucket and all actors.
+
+  - On start up (a 'ready' hook) set the range bands and modifiers based 
+    on the current setting value.
+
+  - On start up (a 'ready' hook) initializes the range ruler to show range penalties
+    according to the chosen strategy.
+    
+  - Maintains an instance variable (rangeTable) that contains the current set of
+    range bands based on the chosen strategy. This table is to be used only for display
+    and interactive roll mod application purposes
+
+  - Maintains an instance variable (bucketModifiers) that contains an array of
+    modifier text for the modifier bucket.
+ */
+export default class GURPSRange {
+    constructor() {
+        this._setup();
+        this._setStrategy(STRATEGY_STANDARD);
+    }
+
+    _setup() {
+        let self = this;
+
+        Hooks.once('init', async function () {
+            game.GURPS = GURPS;
+
+            const choices = {};
+            choices[STRATEGY_STANDARD] = 'Size and Speed/Range Table';
+            choices[STRATEGY_RANGE_BANDS] = 'Monster Hunters 2 Range Bands';
+
+            /* Define Settings */
+            game.settings.register(SYSTEM_NAME, SETTING_NAME, {
+                name: 'Default range modifier strategy:',
+                hint: 'Sets the formula to use to calculate range penalties.',
+                scope: 'world',
+                config: true,
+                type: String,
+                choices,
+                default: STRATEGY_STANDARD,
```

Comment:
Always use constants instead of strings/numbers if they represent a distinct concept which is referred to in multiple places.